### PR TITLE
cleanup: Lotus client:  remove markets and deal-making from Lotus Client

### PR DESCRIPTION
**Pull request by aarshkshah1992**
## Related Issues
Closes https://github.com/filecoin-project/lotus/issues/12002

## Proposed Changes
The use of legacy markets through Lotus was deprecated a long time ago. This PR completely remove the legacy markets code from the Lotus Client CLI & API. Users can continue using Boost for deal making/as a deal making client.

Specifically, this PR removes the following APIs from the Lotus Full Node:
```
        // ClientImport imports file under the specified path into filestore.
	ClientImport(ctx context.Context, ref FileRef) (*ImportRes, error) //perm:admin
	// ClientRemoveImport removes file import
	ClientRemoveImport(ctx context.Context, importID imports.ID) error //perm:admin
	// ClientStartDeal proposes a deal with a miner.
	ClientStartDeal(ctx context.Context, params *StartDealParams) (*cid.Cid, error) //perm:admin
	// ClientStatelessDeal fire-and-forget-proposes an offline deal to a miner without subsequent tracking.
	ClientStatelessDeal(ctx context.Context, params *StartDealParams) (*cid.Cid, error) //perm:write
	// ClientGetDealInfo returns the latest information about a given deal.
	ClientGetDealInfo(context.Context, cid.Cid) (*DealInfo, error) //perm:read
	// ClientListDeals returns information about the deals made by the local client.
	ClientListDeals(ctx context.Context) ([]DealInfo, error) //perm:write
	// ClientGetDealUpdates returns the status of updated deals
	ClientGetDealUpdates(ctx context.Context) (<-chan DealInfo, error) //perm:write
	// ClientGetDealStatus returns status given a code
	ClientGetDealStatus(ctx context.Context, statusCode uint64) (string, error) //perm:read
	// ClientHasLocal indicates whether a certain CID is locally stored.
	ClientHasLocal(ctx context.Context, root cid.Cid) (bool, error) //perm:write
	// ClientFindData identifies peers that have a certain file, and returns QueryOffers (one per peer).
	ClientFindData(ctx context.Context, root cid.Cid, piece *cid.Cid) ([]QueryOffer, error) //perm:read
	// ClientMinerQueryOffer returns a QueryOffer for the specific miner and file.
	ClientMinerQueryOffer(ctx context.Context, miner address.Address, root cid.Cid, piece *cid.Cid) (QueryOffer, error) //perm:read
	// ClientRetrieve initiates the retrieval of a file, as specified in the order.
	ClientRetrieve(ctx context.Context, params RetrievalOrder) (*RestrievalRes, error) //perm:admin
	// ClientRetrieveWait waits for retrieval to be complete
	ClientRetrieveWait(ctx context.Context, deal retrievalmarket.DealID) error //perm:admin
	// ClientExport exports a file stored in the local filestore to a system file
	ClientExport(ctx context.Context, exportRef ExportRef, fileRef FileRef) error //perm:admin
	// ClientListRetrievals returns information about retrievals made by the local client
	ClientListRetrievals(ctx context.Context) ([]RetrievalInfo, error) //perm:write
	// ClientGetRetrievalUpdates returns status of updated retrieval deals
	ClientGetRetrievalUpdates(ctx context.Context) (<-chan RetrievalInfo, error) //perm:write
	// ClientQueryAsk returns a signed StorageAsk from the specified miner.
	ClientQueryAsk(ctx context.Context, p peer.ID, miner address.Address) (*StorageAsk, error) //perm:read
	// ClientCalcCommP calculates the CommP and data size of the specified CID
	ClientDealPieceCID(ctx context.Context, root cid.Cid) (DataCIDSize, error) //perm:read
	// ClientCalcCommP calculates the CommP for a specified file
	ClientCalcCommP(ctx context.Context, inpath string) (*CommPRet, error) //perm:write
	// ClientGenCar generates a CAR file for the specified file.
	ClientGenCar(ctx context.Context, ref FileRef, outpath string) error //perm:write
	// ClientDealSize calculates real deal data size
	ClientDealSize(ctx context.Context, root cid.Cid) (DataSize, error) //perm:read
	// ClientListTransfers returns the status of all ongoing transfers of data
	ClientListDataTransfers(ctx context.Context) ([]DataTransferChannel, error)        //perm:write
	ClientDataTransferUpdates(ctx context.Context) (<-chan DataTransferChannel, error) //perm:write
	// ClientRestartDataTransfer attempts to restart a data transfer with the given transfer ID and other peer
	ClientRestartDataTransfer(ctx context.Context, transferID datatransfer.TransferID, otherPeer peer.ID, isInitiator bool) error //perm:write
	// ClientCancelDataTransfer cancels a data transfer with the given transfer ID and other peer
	ClientCancelDataTransfer(ctx context.Context, transferID datatransfer.TransferID, otherPeer peer.ID, isInitiator bool) error //perm:write
	// ClientRetrieveTryRestartInsufficientFunds attempts to restart stalled retrievals on a given payment channel
	// which are stuck due to insufficient funds
	ClientRetrieveTryRestartInsufficientFunds(ctx context.Context, paymentChannel address.Address) error //perm:write

	// ClientCancelRetrievalDeal cancels an ongoing retrieval deal based on DealID
	ClientCancelRetrievalDeal(ctx context.Context, dealid retrievalmarket.DealID) error //perm:write

	// ClientUnimport removes references to the specified file from filestore
	// ClientUnimport(path string)

	// ClientListImports lists imported files and their root CIDs
	ClientListImports(ctx context.Context) ([]Import, error) //perm:write
```

**_In addition to the above_**, the `lotus client` CLI family of commands has also been completely removed.

And  also removed the following CLI commands from Lotus:

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] If the PR affects users (e.g., new feature, bug fix, system requirements change), update the CHANGELOG.md and add details to the UNRELEASED section.
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

**Comment by rvagg** [comment]
I guess we've been warning about this for a while, I only noticed this today on some CI output:

```
2024-05-16T07:30:55.950Z	ERROR	alerting	alerting/alerts.go:106	alert raised	{"type": {"System":"system","Subsystem":"EOL"}, "message": {"message":"The lotus-miner legacy markets subsystem is deprecated and will be removed in a future release. Please migrate to [Boost](https://boost.filecoin.io/) or similar markets subsystems."}}
```

**Comment by LexLuthr** [comment]
From Boost point of view this looks Good. Boost switched to it's own client a while ago for testing. We do have a full suite of deal tests there. Lotus client has been unsupported by Boost for a while. 
Curio will have a new tests for deals which will probably skip using any client. 
Magik can confirm about lotus-miner as itest suite is very tightly coupled with client API I think.

**Comment by rvagg** [comment]
@aarshkshah1992 I'll have to look at this tomorrow, but further to @ribasushi's point - would you be able to make a basic (rough) inventory of which parts of builtin actors behaviours/methods we _may_ be leaving with less coverage here in the Lotus repo. While it's good to know that Boost and Curio will have coverage, it's important that we also maintain enough coverage so we have confidence to iterate on actors while knowing they're going to be exercised properly. We don't necessarily have to keep tests around, but it might give us a good idea of what we need to focus on build separate to the deal flow (i.e. more manually than these tests do).

**Comment by aarshkshah1992** [comment]
Will make the adjustments to https://github.com/filecoin-project/lotus/blob/master/node/config/def.go in a subsequent PR that I am raising for the provider side of things.

**Comment by aarshkshah1992** [comment]
Follow-up PR to remove Markets from Lotus Miner and Full Node is at https://github.com/filecoin-project/lotus/pull/12005.

**Comment by aarshkshah1992** [comment]
@rvagg @ribasushi Yes will do a write-up on what test coverage we're lacking now.

**Review by ZenGround0 — APPROVED** [review]

File: api/api_full.go
Comment (ZenGround0) [reply api/api_full.go:392]: * I don't think they *need* to use go-fil-markets.  But if its hard to recreate without the API call I can do it later.

Comment (ZenGround0) [reply api/api_full.go:392]: idk because I'm a phillistine who just wants to play with his deal smart contracts

Comment (ZenGround0) [reply api/api_full.go:392]: Me have file me want commp

Comment (rvagg) [reply api/api_full.go:392]: yeah, original discussion on this suggested that we may want to move some things to lotus-shed, and commp seems reasonable

`boostx` has all this functionality if you want to install it

in terms of unixfs generation, just use go-car, `car` is not a bad little utility to have installed, and then one of the existing commp generation utilities like `stream-commp`

Comment (aarshkshah1992) [reply api/api_full.go:392]: Yeah I am not sure we need to keep the `CommP` functionality around in lotus-shed given the plethora of other tools out there that already do it if Lotus itself does not need this functionality for anything anymore.

Comment (jennijuju) [reply api/api_full.go:392]: Worth checking the docs/tutorials - pretty sure it’s referenced in a couple places

**Comment by aarshkshah1992** [comment]
@rvagg I went through the itests we have deleted and it dosen't look like we're losing any meaningful Actors coverage:

1. The DDO itests you wrote already cover deal publishing and waiting for sectors to be sealed and they are the main code paths getting removed from the deleted tests (that concern us).

2.  Other than that, we're losing coverage of some Miner/Miner-Worker behaviour wrt to sector upgrades and sector upgrade aborts but that should be covered by Curio.

File: itests/worker_upgrade_test.go
Comment (aarshkshah1992) [inline itests/worker_upgrade_test.go:19]: I think we need a test for upgrading CC sectors.

Comment (aarshkshah1992) [reply itests/worker_upgrade_test.go:19]: This will get covered as part of https://github.com/filecoin-project/lotus/pull/12017.

File: chain/sub/incoming.go
Comment (rvagg) [inline chain/sub/incoming.go:38]:
```suggestion
```

File: cmd/lotus-miner/sectors.go
Comment (rvagg) [inline cmd/lotus-miner/sectors.go:591]: I don't think this section should be removed, it's not markets related but just happens to use `GetMarketsAPI` to get the `StorageMinerAPI`. Replace it with this and it should work:

```go
		nodeApi, closer, err := lcli.GetStorageMinerAPI(cctx)
```

File: documentation/en/cli-lotus-miner.md
Comment (rvagg) [inline documentation/en/cli-lotus-miner.md:601]: this should get restored with docsgen-cli after the `sectors refs` cmd restoration

File: itests/path_type_filters_test.go
Comment (rvagg) [inline itests/path_type_filters_test.go:43]: oh, this test seems like a bit of a loss all because it needs deals to exercise storage; it'd be good to get some curio ppl feedback on this one

File: itests/sector_finalize_early_test.go
Comment (rvagg) [inline itests/sector_finalize_early_test.go:60]: another test that's in the unfortunate blast radius; I wonder if we could replace this functionality with something simpler so this doesn't all have to go away

File: itests/worker_test.go
Comment (rvagg) [inline itests/worker_test.go:616]: ditto; would be good to not lose this one too

**Comment by rvagg** [comment]
This looks great to me, exciting to remove so much legacy.

2 things to resolve: removal of lib/unixfs package, and restoration of `sectors refs` CLI.

And there's the few tests in there that are not deal related but get caught up in the lack of deal infrastructure, yet they're testing important functionality that leaves a gap.

I think next step might be to get an eye from @magik6k; both in terms of the general approach here but also for some feedback about those tests that I marked as a concern—should we put some work in to replacing the deal infrastructure with some data onboarding flows to keep the tests active?

File: chain/sub/incoming.go
Comment (rvagg) [reply chain/sub/incoming.go:43]: oh, I see this is taken care of in #12005! nice

File: cmd/lotus-miner/sectors.go
Comment (aarshkshah1992) [reply cmd/lotus-miner/sectors.go:591]: Done

File: itests/worker_test.go
Comment (aarshkshah1992) [reply itests/worker_test.go:616]: I mean this should get covered by Curio tests.

File: chain/sub/incoming.go
Comment (aarshkshah1992) [reply chain/sub/incoming.go:43]: Yeah this is taken care of

File: cmd/lotus-miner/sectors.go
Comment (aarshkshah1992) [reply cmd/lotus-miner/sectors.go:591]: You are right, fixed this.

File: documentation/en/cli-lotus-miner.md
Comment (aarshkshah1992) [reply documentation/en/cli-lotus-miner.md:601]: You are right, fixed this.

File: itests/path_type_filters_test.go
Comment (aarshkshah1992) [reply itests/path_type_filters_test.go:43]: I think this should get covered by Curio tests. I am not sure if it is great ROI building a lot of testing deal onboarding infra just for this.

**Comment by aarshkshah1992** [comment]
@rvagg I am not sure if it is high ROI to add and then maintain deal onboarding code just to make up for a couple of tests. Some of that should be covered by Curio tests.

And we're adding tests for SnapDeals and DDO to cover the Actors interaction as part of our NI-PoRep testing work.

**Comment by rvagg** [comment]
lgtm! if we can get sign-off from a miner expert then this should be gtg

**Pull request merged**
